tests/composition/composition-grid-layout.spec.js

```diff
@@ -38,30 +38,47 @@ test.afterEach(async ({ page }, testInfo) => {
   await updateTestResults(testInfo.status, testInfo.retry)
 });
 
-mainTest(qase([1689,1696],'PENPOT-1689,1696 Check grid lines, check edit mode in the right panel'), async ({ page }) => {
-  await mainPage.createDefaultBoardByCoordinates(200, 300);
-  await designPanelPage.changeHeightAndWidthForLayer('300', '400');
-  await mainPage.waitForChangeIsSaved();
-  await mainPage.addGridLayoutViaRightClick();
-  await mainPage.waitForChangeIsUnsaved();
-  await mainPage.waitForChangeIsSaved();
-  await designPanelPage.isLayoutRemoveButtonExists();
-  await mainPage.clickViewportOnce();
-  await mainPage.clickCreatedBoardTitleOnCanvas();
-  await expect(mainPage.viewport).toHaveScreenshot('board-with-grid-layout.png', {
-    mask: [mainPage.guides, mainPage.guidesFragment, mainPage.toolBarWindow]
+mainTest.describe(() => {
+  mainTest.beforeEach(async ({ page, browserName }, testInfo) => {
+    await mainPage.createDefaultBoardByCoordinates(200, 300);
+    await designPanelPage.changeHeightAndWidthForLayer('300', '400');
+    await mainPage.waitForChangeIsSaved();
+    await mainPage.addGridLayoutViaRightClick();
+    await mainPage.waitForChangeIsUnsaved();
+    await mainPage.waitForChangeIsSaved();
+    await designPanelPage.isLayoutRemoveButtonExists();
+    await mainPage.clickViewportOnce();
+    await mainPage.clickCreatedBoardTitleOnCanvas();
+  });
+
+  mainTest(qase([1689,1696],'PENPOT-1689,1696 Check grid lines, check edit mode in the right panel'), async () => {
+    await expect(mainPage.viewport).toHaveScreenshot('board-with-grid-layout.png', {
+      mask: [mainPage.guides, mainPage.guidesFragment, mainPage.toolBarWindow]
+    });
+    await designPanelPage.openGridEditModeFromDesignPanel();
+    await mainPage.waitForChangeIsSaved();
+    await expect(mainPage.viewport).toHaveScreenshot('board-with-grid-edit-mode.png', {
+      mask: [mainPage.guides, mainPage.guidesFragment]
+    });
+    await expect(mainPage.fileRightSidebarAside).toHaveScreenshot(
+      'grid-edit-right-sidebar-image.png',
+      {
+        mask: [mainPage.usersSection],
+      },
+    );
+  });
+
+  mainTest(qase(1863,'Add a lot of columns and check "Columns" panel'), async () => {
+    await designPanelPage.openGridEditModeFromDesignPanel();
+    await mainPage.waitForChangeIsSaved();
+    await mainPage.addColumnsGridLayout(30);
+    await mainPage.waitForChangeIsSaved();
+    await expect(mainPage.viewport).toHaveScreenshot('board-with-grid-3-30.png', {
+      mask: [mainPage.guides],
+    });
+    await expect(mainPage.columnsRowsOnDesignPanel).toHaveScreenshot('lot-1fr-columns.png');
+
```

Comment:
Do you think it's possible to test this without screenshots? I'm trying to imagine a future of automated testing that avoids screenshots in most cases.

Reply:
The screenshot is necessary because we are verifying that the text is truncated, and this truncation is only visible visually.
![image](https://github.com/user-attachments/assets/83f886fa-ec0a-4ec9-8926-b2f084010efc)
![image](https://github.com/user-attachments/assets/0ed35ed7-57d8-4b12-909c-a595fe57253c)

Reply:
Ok!